### PR TITLE
CM-414: Created Containerfile for istio-csr with redhat base image.

### PR DESCRIPTION
**Pull request by anirudhAgniRedhat**
- Added submodule for [istio-csr](https://github.com/openshift/cert-manager-istio-csr)
- Added containerfile for istio-csr

### CODE REVIEW

File: hack/shell-scripts-linter.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 verify_script()
 {
-	if ! find . -type f -name '*.sh' '!' -path './cert-manager/*' '!' -path './cert-manager-operator/*' \
+	if ! find . -type f -name '*.sh' '!' -path './cert-manager/*' '!' -path './cert-manager-operator/*' -path './istio-csr/*' '!' \
```
Comment (bharath-b-rh) [inline hack/shell-scripts-linter.sh:5]:
```suggestion
	if ! find . -type f -name '*.sh' '!' -path './cert-manager/*' '!' -path './cert-manager-operator/*' '!' -path './istio-csr/*' \
```

File: hack/containerfile-linter.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 declare -a CONTAINERFILES
 
-CERT_MANAGER_OPERATOR_CONTAINERFILES=("Containerfile.cert-manager" "Containerfile.cert-manager.acmesolver" "Containerfile.cert-manager-operator" "Containerfile.cert-manager-operator.bundle")
+CERT_MANAGER_OPERATOR_CONTAINERFILES=("Containerfile.cert-manager" "Containerfile.cert-manager.acmesolver" "Containerfile.cert-manager-operator" "Containerfile.cert-manager-operator.bundle" "Containerfile.istio-csr")
```
Comment (bharath-b-rh) [inline hack/containerfile-linter.sh:5]: Not sure why each file was listed earlier. Please update it to

```suggestion
CERT_MANAGER_OPERATOR_CONTAINERFILES=($(ls Containerfile.*))
```

File: Makefile [hunk 3]
```diff
@@ -31,6 +33,12 @@ ifeq ($(PARENT_BRANCH_SUFFIX), main)
 CERT_MANAGER_OPERATOR_BRANCH = master
 endif
 
+ISTIO_PARENT_BRANCH = $(strip $(shell git branch --show-current))
+
+ISTIO_BRANCH_SUFFIX = $(if $(filter release-%, $(ISTIO_PARENT_BRANCH)),$(strip $(shell echo $(ISTIO_PARENT_BRANCH) | cut -d'-' -f2)),main)
+
+ISTIO_CSR_BRANCH ?= $(if $(filter release-%, $(ISTIO_PARENT_BRANCH)),release-$(ISTIO_BRANCH_SUFFIX),main)
+
```
Comment (bharath-b-rh) [inline Makefile:41]: istio-csr repository too will have release branches, ex `release-1.15`, we won't be needing the changes related to branch name parsing.

File: .gitmodules [hunk 4]
```diff
@@ -6,3 +6,6 @@
 	path = cert-manager-operator
 	url = https://github.com/openshift/cert-manager-operator.git
 	branch = master
+[submodule "istio-csr"]
+	path = istio-csr
+	url = https://github.com/openshift/cert-manager-istio-csr
```
Comment (bharath-b-rh) [inline .gitmodules:11]: Please set branch for the sub-module using `git submodule set-branch --branch main istio-csr` command.
Sorry missed this input.

Comment (anirudhAgniRedhat) [reply .gitmodules:11]: Added Thanks!

File: Makefile [hunk 5]
```diff
@@ -111,6 +125,7 @@ build-operator-image:
 build-operand-images:
 	$(IMAGE_BUILD_CMD) -f $(cert_manager_containerfile_name) -t $(CERT_MANAGER_IMAGE):$(IMAGE_VERSION) .
 	$(IMAGE_BUILD_CMD) -f $(cert_manager_acmesolver_containerfile_name) -t $(CERT_MANAGER_ACMESOLVER_IMAGE):$(IMAGE_VERSION) .
+	$(IMAGE_BUILD_CMD) -f $(istio_csr_containerfile_name) -t $(ISTIO_CSR_IMAGE):$(IMAGE_VERSION) .
```
Comment (bharath-b-rh) [inline Makefile:128]: nit: an improvement to existing code.
```suggestion
build-operand-images: build-cert-manager-image build-cert-manager-acmesolver-image build-istio-csr-image
```

**Comment by bharath-b-rh** [comment]
Please incorporate last review suggestion.
/lgtm

File: Makefile (hunk 5 above)
Comment (anirudhAgniRedhat) [reply Makefile]: Thanks for the suggestion added!!

File: Containerfile.istio-csr [hunk 6]
```diff
@@ -0,0 +1,43 @@
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_golang_1.22 AS builder
```
Comment (arkadeepsen) [inline Containerfile.istio-csr:1]: @bharath-b-rh Shall we name the file `Containerfile.cert-manager-istio-csr` to be consistent with the repo name?

Comment (bharath-b-rh) [reply Containerfile.istio-csr:1]: Yeah +1

Comment (anirudhAgniRedhat) [reply Containerfile.istio-csr:1]: Should we change name of image as well to https://github.com/openshift/cert-manager-operator-release/pull/94/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R62 `cert-manager-istio-csr` ?

Comment (anirudhAgniRedhat) [reply Containerfile.istio-csr:1]: also submodule name to same convention?

Comment (bharath-b-rh) [reply Containerfile.istio-csr:1]: yeah, sounds good.

**Pull request merged**